bat-utils/lib/runtime-wallet.js

```diff
@@ -475,4 +496,118 @@ Wallet.providers.mock = {
 }
 Wallet.providers.mockHttpSignature = Wallet.providers.mock
 
+Wallet.providers.simplex = {
+  purchaseBAT: async function (info, amount, currency, language) {
+    const fiat = 'USD'
+    const now = underscore.now()
+    const params = {
+      currency: 'XBT',
+      partner: 'brave',
+      version: '1'
+    }
+    let expires, quote, rate, result
+
+    if (!this.runtime.config.simplex) return
+
+    if (!this.currency.fiatP(currency)) {
+      rate = underscore.pick(this.currency.rates[currency] || {}, fiat)
+      if (!rate) return
+
+      amount = Math.ceil(this.currency.alt2fiat(currency, amount, fiat, true))
+      currency = fiat
+    }
+
+    quote = info.simplex
+    if ((!quote) || (quote.expires <= now)) {
+      result = await braveHapi.wreck.post(this.runtime.config.simplex.url + '/wallet/merchant/v2/quote', {
+        headers: {
+          accept: 'application/json',
+          authorization: 'ApiKey ' + this.runtime.config.simplex.api_key,
+          'content-type': 'application/json'
+        },
+        payload: JSON.stringify({
+          end_user_id: info.paymentId,
+          digital_currency: params.currency,
+          fiat_currency: currency,
+          requested_currency: currency,
+          requested_amount: amount,
+          wallet_id: params.partner
+        })
+      })
+      if (Buffer.isBuffer(result)) try { result = JSON.parse(result) } catch (ex) { result = result.toString() }
+
+      expires = new Date(result.valid_until).getTime()
+      if ((isNaN(expires)) || (!result.quote_id)) return
+
+      quote = underscore.extend(result, { expires: expires })
+    }
+    quote.payment_id = uuid.v4().toLowerCase()
+
+    result = await braveHapi.wreck.post(this.runtime.config.simplex.url + '/wallet/merchant/v1/payments/partner/data', {
+      headers: {
+        accept: 'application/json',
+        authorization: 'ApiKey ' + this.runtime.config.simplex.api_key,
+        'content-type': 'application/json'
+      },
+      payload: JSON.stringify({
+        account_details: {
+          app_provider_id: params.partner,
+          app_version_id: params.version,
+          app_end_user_id: quote.user_id,
+          signup_login: {
+            ip: '213.162.55.5',
```

Comment:
what is the significance of this ip?

Reply:
the significance is that it is a non-existent IP address, cf., https://tools.ietf.org/html/rfc1918

we aren't going to supply the browser's IP address to the provider.

Reply:
sure, the browser will do that itself when it follows the link to go to the provider page :p 

I was wondering in the context of simplex but based on your answer it's the expected ip of the user signing up and can be wrong / spoofed without issue.